app/forms/BusinessContactDetailsFormProvider.scala

```diff
@@ -31,35 +31,54 @@ import models.BusinessContactDetails
 
 class BusinessContactDetailsFormProvider @Inject() extends Mappings {
 
-  private val util                          = PhoneNumberUtil.getInstance
-  def apply(): Form[BusinessContactDetails] =
-    Form(
-      mapping(
-        "name"  -> text(nameRequiredError)
-          .verifying(Constraints.pattern(Validation.nameInputPattern, error = nameFormatError))
-          .verifying(maxLength(Validation.nameMaxLength, nameLengthError)),
-        "email" -> text(emailRequiredError)
-          .verifying(maxLength(Validation.emailMaxLength, emailLengthError))
-          .verifying(Constraints.pattern(Validation.emailPattern, error = emailFormatError)),
-        "phone" -> text(phoneRequiredError)
-          .verifying(
-            phoneFormatError,
-            phone =>
-              isValid(phone)
-                && phone.length <= Validation.phoneNumberMaxLength
-                && !phone.exists(_.isLetter)
-          )
-      )(BusinessContactDetails.apply)(
-        (businessContactDetails: BusinessContactDetails) =>
-          Some(
-            (
-              businessContactDetails.name,
-              businessContactDetails.email,
-              businessContactDetails.phone
-            )
-          )
+  private val util = PhoneNumberUtil.getInstance
+
+  val nameMapping: (String, Mapping[String]) = "name" -> text(nameRequiredError)
+    .verifying(Constraints.pattern(Validation.nameInputPattern, error = nameFormatError))
+    .verifying(maxLength(Validation.nameMaxLength, nameLengthError))
+
+  val emailMapping: (String, Mapping[String]) = "email" -> text(emailRequiredError)
+    .verifying(maxLength(Validation.emailMaxLength, emailLengthError))
+    .verifying(Constraints.pattern(Validation.emailPattern, error = emailFormatError))
+
+  val phoneMapping: (String, Mapping[String]) = "phone" -> text(phoneRequiredError)
+    .verifying(phoneFormatError, phone => isValid(phone))
+    .verifying(phoneLengthError, phone => phone.length <= Validation.phoneNumberMaxLength)
+    .verifying(phoneLetterError, phone => !phone.exists(_.isLetter))
+
+  val companyNameMapping: (String, Mapping[String]) =
+    "companyName" -> text(companyNameRequiredError)
+
+  val defaultMap = mapping(nameMapping, emailMapping, phoneMapping)(
+    (name, email, phone) => BusinessContactDetails.apply(name, email, phone, None)
+  )(
+    (businessContactDetails: BusinessContactDetails) =>
+      Some(
+        (
+          businessContactDetails.name,
+          businessContactDetails.email,
+          businessContactDetails.phone
+        )
       )
-    )
+  )
+
+  val companyNameIncMap = mapping(nameMapping, emailMapping, phoneMapping, companyNameMapping)(
+    (name, email, phone, companyName) =>
+      BusinessContactDetails.apply(name, email, phone, Some(companyName))
```

Comment:
Yes this is probably the neatest way we can work around that fixed number of elements in the mapping....